### PR TITLE
Add a toggle to use context attributes on the regex policy provider

### LINKED ISSUE #23470: Docs: authorization_services/topics/service-authorization-obtaining-permission.adoc
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

authorization-services

### Describe the bug

File: authorization_services/topics/service-authorization-obtaining-permission.adoc

In the "Obtaining permissions" chapter, there is an optional field name "claim_token", according to the document it is "A string representing additional claims that should be considered by the server when evaluating permissions for the resource(s) and scope(s) being requested"

So basically, by using a regex policy, I would have expected to been able to check the string provided in the claim_token matche with the regex. But unfortunately, it seems that regex policy only look at user attribute.

This stackoverflow post is related to the problem we're facing:
https://stackoverflow.com/questions/75855902/pushing-claims-to-regex-based-policy-in-keycloak

### Version

22.0.3

### Expected behavior

How to use a "claim_token" in a regex policy ?

### Actual behavior

regex policy is only using user attribute

### How to Reproduce?

- create a regex policy with he following params:
  name: destination
  target claim: destination
  regex pattern: ^desk.+$

- Send to the request to KC, with the following params:
  "claim_token_format": "urn:ietf:params:oauth:token-type:jwt" 
  "claim_token": '{"destination":["deskA"]}' // encoded in base64

permission using the "destination" policy is always denied (unless we add a "destination" attribute to the user). In any case, the value from "claim_token" is not used to evaluate the regex

### Anything else?

- Is there a setting to do in KC to use the claim_token value in regexp policy ?

- Do we have to create a custom "regex policy" looking at the `Attributes attributes = evaluation.getContext().getAttributes();` (claim_token value) ?

- any other suggestion ?

### PR DESCRIPTION
**Pull request by jlsjonas**
Allows (explicitly, for security reasons) targeting context claims with the regex policy

Closes #23470

### CODE REVIEW

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:60]: The `fieldId` property needs to match the `id` attribute of the the form controls, in this case the `Checkbox`. There is also already a control with the id `pattern`, so this should not be re-used.

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:64]: You should not add namespaces here.
```suggestion
            fieldLabelId="targetContextAttributes"
```

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:67]: Should this really be a required field? A checkbox already has a value at all times.

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:77]: This `id` should match the name of the field in the form.

```suggestion
              id="targetContextAttributes"
```

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:78]: I don't believe this test id is used anywhere, but the name should probably match the conventions of the rest of the component.

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:79]: There is no need to specify a `name` for this field, this is already handled by React Form Hook.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (remituveri) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: renamed "targetContextAttributes"
Notes: I've tested it locally, and in fact the Checkbox Id does not matter, but the Controller name has to match with the new name "targetContextAttributes"

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (remituveri) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: done

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (remituveri) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: no, indeed, it was not required

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (remituveri) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: removed

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (remituveri) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: correct! the chackbox's name has been removed

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (remituveri) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: checkbox's id does not matter but the controller name has to match with the fieldId of the formGroup. (tested locally)

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: The checkbox `id` needs to match, as otherwise the input is not associated with the label. This creates problems for visually impaired users, as their accessibility tools are not able to understand the structure. It also handles other behavior, such as focusing the input when the label is clicked.

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [inline js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx:74]: The `id` prop passed to `Checkbox` needs to match the one provided as `fieldId` to the `FormGroup`, as otherwise the input is not associated with the label.

This creates problems for visually impaired users, as their accessibility tools are not able to understand the structure. It also handles other behavior, such as focusing the input when the label is clicked.

```suggestion
              id="targetContextAttributes"
```

File: js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx
Comment (jonkoops) [reply js/apps/admin-ui/src/clients/authorization/policy/Regex.tsx]: It does, closing this as resolved, but see my new review comment.

**Review by jonkoops — DISMISSED** [review]
Looks good to me from a front-end perspective, whether the Java changes are valid and wanted I will leave up to another maintainer.

**Comment by remituveri** [comment]
@ssilvert 
have you had the chance to take a look at this PR ?

**Comment by remituveri** [comment]
Hello @jonkoops , there is no movement on this PR since you approved it. Is there something we can do to move forward with it ? BR

**Comment by jonkoops** [comment]
We need someone from the core team to take a look at this to see if this functionality we want to introduce. cc @keycloak/core

**Comment by jonkoops** [comment]
@remituveri Also I'd recommend fixing the merge conflicts if you want to get this merged.

**Review by pedroigor — DISMISSED** [review]
@jlsjonas Sorry for the delay. Also LGTM.

Would you mind rebasing?

**Comment by remituveri** [comment]
@pedroigor rebasing has been performed. Then I've noticed that the translation.json file for admin ui does not exist anymore so I've added the "translation" in js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties. not sure if it's the good way to do it..

**Review by pedroigor — APPROVED** [review]
@jonkoops Do you mind re-reviewing the UI-related changes.

**Comment by pedroigor** [comment]
@jlsjonas Thanks again. If tests are passing, LGTM.

**Comment by pedroigor** [comment]
@jlsjonas Now we need a sign-off commit message and for that we probably want to squash all commits to make it easier :)

**Comment by remituveri** [comment]
@pedroigor , Voila, it's all yours.

**Review by jonkoops — APPROVED** [review]

**Review by pedroigor — APPROVED** [review]

**Pull request merged**
